### ISSUE TITLE
*cough cough* why not ditch react?

### ISSUE THREAD

**Issue by 9ebd7134** [blueedgetechno/windows11#25]
**Is your feature request related to a problem? Please describe.**
I hate react, its too slow. (my opinion don't fight)

**Describe the solution you'd like**
I'd do it myself, Lets just ditch react to make the thing faster shall we?

**Describe alternatives you've considered**
VANILA JAVASCRIPT

**Additional context**
Don't fight me :)

Thanks hope you allow me to ditch stupid react and make life better for all the human beings in the world

**Comment by blueedgetechno** [comment]
I will fight XD. React might be slow but very powerful. It gives to access to the endless open-source libraries and a fully functional application on the cost of performance, which can be reduced drastically.

And Redux on the top of that is like a cake you can't refuse.

**Comment by 9ebd7134** [comment]
So your telling me vanilla js is less powerful ? im pretty sure with it being faster you also get the benefit of it having less bugs :)

**Comment by 9ebd7134** [comment]
Also sites slower than 0.3ms are not the best :) since many uesrs click off

**Comment by blueedgetechno** [comment]
I know but the project was not initially intended to attract users. Anyone with decent network and patience can make through the delay. Also I don't how much effort it will take to remove the react dependency from the project and replace it with vanilla.

**Comment by 9ebd7134** [comment]
if you allow me to do this, i won't refactor the project i'll make a new project

**Comment by blueedgetechno** [comment]
Yeah, you can but then I will have to spend another few months learning vanilla JS in order to contribute.

**Comment by heyarviind** [comment]
> Yeah, you can but then I will have to spend another few months learning vanilla JS in order to contribute.

Whoops, so you learned react without learning javascript?

**Comment by blueedgetechno** [comment]
There are people who spend time mastering javascript before anything else and there are some ... those ... who ...

**Comment by khang-nd** [comment]
@9ebd7134 good luck making one with Vanilla JS, you will need it... a lot.

**Comment by 9ebd7134** [comment]
> > Yeah, you can but then I will have to spend another few months learning vanilla JS in order to contribute.
> 
> Whoops, so you learned react without learning javascript?

wanted to say that myself :)

**Comment by 9ebd7134** [comment]
shoot, i forgot to mention I am going to use ts

**Comment by 9ebd7134** [comment]
were did y9ou get the icons from ? @blueedgetechno

**Comment by blueedgetechno** [comment]
@9ebd7134 I scrapped the system32 folders and extracted icons from exe files

**Comment by 9ebd7134** [comment]
oh :+1:

**Comment by hktr92** [comment]
> Noticed it was abnormally slow on Chrome/Linux (and interestingly faster on Firefox/Linux) for this PC: Ryzen 7 3700X / 16GB RAM / Nvidia RTX 2700 / Chrome 88 / Firefox 86
> 
> Discussing performance between React/vdom vs JS, https://macos.vercel.app/ is pretty fun to check out as well. It's funny how this brings the parallel: [Win11 in React vs MacOS in Svelte](https://www.youtube.com/watch?v=Jkrn6ecxthM)

anyone wants to make a plasma 5 in vue? or gnome 3 in angular?

**Comment by Lukas-Batema** [comment]
> **Is your feature request related to a problem? Please describe.**
> 
> I hate react, its too slow. (my opinion don't fight)
> 
> 
> 
> **Describe the solution you'd like**
> 
> I'd do it myself, Lets just ditch react to make the thing faster shall we?
> 
> 
> 
> **Describe alternatives you've considered**
> 
> VANILA JAVASCRIPT
> 
> 
> 
> **Additional context**
> 
> Don't fight me :)
> 
> 
> 
> Thanks hope you allow me to ditch stupid react and make life better for all the human beings in the world
> 
> 

*cough cough* Why not switch to jQuery? *cough cough*

**Comment by blueedgetechno** [comment]
_cough cough_ why are we coughing _cough cough_

**Comment by Lukas-Batema** [comment]
> _cough cough_ why are we coughing _cough cough_

*sneeze sneeze* Is this better? *sneeze sneeze*

**Comment by Lukas-Batema** [comment]
Nice labelling... `uhh...`

**Comment by Lukas-Batema** [comment]
In all seriousness, React takes complicated scripts to set up something that you can do, in some cases, with normal JS, or even jQuery (yes, I know that's a very overused library, but you know, if it works, it works). That is why I am going to rewrite this soon with all vanilla JS and some jQuery.

**Comment by PhantomRex123** [comment]
The issue isn't with js its with html/css. if anyone is willing to ditch react tell me.

**Comment by Lukas-Batema** [comment]
> The issue isn't with js its with html/css. if anyone is willing to ditch react tell me.

The issue is with the JS, as mentioned in the original post. Also, if we change from React, we would have to modify some or most of the HTML, in some way, shape, or form. Lastly, I am all for ditching React, but I just wanted to point out that what your original post said, was taken (at least by me), that JS/React was the problem, not the HTML. Hopefully you understand. 

>_ Best Wishes,
> Lukas Batema

**Comment by PhantomRex123** [comment]
Sorry for the reply time, i never said the problem was with the html, i get that React's html/css isn't the same as standard  Html/Css, nevertheless, don't think react would help in this project in any way or sort. This is purely based on styling, and is simply dependency mess.

**Comment by Lukas-Batema** [comment]
> The issue isn't with js its with html/css. if anyone is willing to ditch react tell me.

I beg your pardon?

**Comment by PhantomRex123** [comment]
> > The issue isn't with js its with html/css. if anyone is willing to ditch react tell me.
> 
> I beg your pardon?

Sorry, 

Nevertheless i said that since im awful at html/css. i semi created the project, and then realized my css is shit. i got some applications working thought. (Browser, console which was cloned from another project of mine, gnvim, file explorer)

**Comment by Lukas-Batema** [comment]
> > > The issue isn't with js its with html/css. if anyone is willing to ditch react tell me.
> > 
> > 
> > I beg your pardon?
> 
> Sorry,
> 
> Nevertheless i said that since im awful at html/css. i semi created the project, and then realized my css is shit. i got some applications working thought. (Browser, console which was cloned from another project of mine, gnvim, file explorer)

Thank you for the more clear explanation. I understand now. I would love to help, but I do not have time as of the present and near future.
